src/variable_versions/common.rs

```diff
@@ -70,6 +70,12 @@ impl DataNumber {
                 let ip_addr = Ipv6Addr::from(taken);
                 (i, FieldValue::Ip6Addr(ip_addr))
             }
+            FieldDataType::MacAddr => {
+                let (i, taken) = take(6 as usize)(remaining)?;
+                let taken: &[u8; 6] = taken.try_into().unwrap();
```

Comment:
Instead of Unwrap can we return an Error Type for Result?.  I can handle it myself in a follow up PR if you want.

Reply:
Makes sense, I updated PR.